### PR TITLE
created order and order an order again

### PR DESCRIPTION
**Pull request by JuliusChiamaka**

### CODE REVIEW

File: EatGoodNaija.Server/Controllers/FoodOrderController.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        }
+
+        [HttpPost("{customerId}/reorder/{orderId}")]
+        public async Task<ActionResult<ResponseDTO<OrderDto>>> ReorderAsync(string customerId, string orderId)
+        {
+            var response = new ResponseDTO<OrderDto>();
+            try
+            {
+                var order = await _service.ReorderAsync(customerId, orderId);
+                response.StatusCode = 200;
```
Comment (rocklessg) [inline EatGoodNaija.Server/Controllers/FoodOrderController.cs]: Move this response checks to the services. 
Return Ok(response) in your controller. Your controller shouldn't contain too much code

File: EatGoodNaija.Server/Controllers/FoodOrderController.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                response.DisplayMessage = "Order reordered successfully";
+                response.Result = order;
+                return Ok(response);
+            }
+            catch (Exception ex)
```
Comment (rocklessg) [inline EatGoodNaija.Server/Controllers/FoodOrderController.cs]: Remove this try catch. You can handle all error and responses in your services. Controller should just be making a call to the service and return the response.

File: EatGoodNaija.Server/Services/Implementation/FoodOrderService.cs [hunk 3]
```diff
@@ -12,26 +12,86 @@ public FoodOrderService(IFoodOrderRepository repository)
...
         {
-            var order = await _repository.ReorderAsync(customerId, orderId);
-            return new OrderDto
+            var response = new ResponseDTO<OrderDto>();
+            try
             {
-                Id = order.Id,
-                OrderDate = order.OrderDate,
-                Items = order.Items.Select(item => new CartItemDto
+                if (string.IsNullOrEmpty(customerId))
```
Comment (rocklessg) [inline EatGoodNaija.Server/Services/Implementation/FoodOrderService.cs]: You didn't check/validate orderId also;

You can do;

if (string.IsNullOrEmpty(customerId) || string.IsNullOrEmpty(orderId))

Comment (JuliusChiamaka) [reply EatGoodNaija.Server/Services/Implementation/FoodOrderService.cs]: - resolved

**Pull request merged**